src/Crdt.Sample/Models/Word.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Crdt.Core;
+using Crdt.Entities;
+
+namespace Crdt.Sample.Models;
+
+public class Word : IObjectBase<Word>
+{
+    public required string Text { get; set; }
+    public string? Note { get; set; }
+
+    public Guid Id { get; init; }
+    public DateTimeOffset? DeletedAt { get; set; }
+    public Guid? AntonymId { get; set; }
+
+    public Guid[] GetReferences()
+    {
+        return AntonymId is null ? [] : [AntonymId.Value];
+    }
+
+    public void RemoveReference(Guid id, Commit commit)
+    {
+        if (AntonymId == id) AntonymId = null;
```

Comment:
yes we are planing on clearing references on delete so as to maintain referential integrity, otherwise we need to check if the referenced object is deleted every time we follow a reference, Jason and I determined that would be a huge blocker to migrating FLEx. If we didn't care about the reference integrity then we could just use an off the shelf library.

There may be special cases where we don't clear the reference on delete, but that's TBD, maybe comments?